pkg/mq/interface.go

```diff
@@ -0,0 +1,163 @@
+package mq
+
+import "time"
+
+// Interface must be implemented for a message queuer
+type Interface interface {
+	// Connect to message queuer
+	Connect(timeout time.Duration) error
+
+	// Close connection
+	Close() error
+
+	// Publish
+	Publish(subject string, data []byte) error
+	PublishAsync(subject string, data []byte, ah AckHandler) (string, error)
+
+	// Subscribe
+	Subscribe(subject string, mh MsgHandler, opts ...SubscriptionOption) (Subscription, error)
```

Comment:
@bquenin I do think there's value in also having the ability to subscribe to and publish raw bytes, but I'd like an interface like the one I did for etcd storage that uses protocol buffer messages (see [here](https://github.com/appcelerator/amp/blob/master/data/storage/interface.go)). I would prefer that the interface explicitly call that out, ex: `PublishRaw()`, `SubscribeRaw`, and let `Publish()` and `Subscribe` be used for `proto.Message`. Or we can have `PublishProto()` and `SubscribeProto()`. That will certainly be our own default case, but that might make sense if we decide to also add `PublishJSON()`, `SubscribeJSON()`, etc, to support various strongly typed APIs in addition to raw bytes...